docker/bin/publicip

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+
+set -e
+
+if [ -z "${PUBLIC_IP}" ]; then
+    if cmd-exists "ifconfig" && cmd-exists "ipconfig" && cmd-exists "pcregrep"; then
+        PUBLIC_IP=$(ifconfig | pcregrep -M -o '^[^\t:]+:([^\n]|\n\t)*status: active' | egrep -o -m 1 '^[^\t:]+' | xargs ipconfig getifaddr)
+    fi
+    if [ -z "${PUBLIC_IP}" ] && cmd-exists "ip"; then
+        PUBLIC_IP=$(ip addr | grep 'state UP' -A2 | tail -n1 | awk '{print $2}' | cut -f1  -d'/')
```

Comment:
may yield ipv6 addresses. may want to consider something like this:
```sh
ip -o -f inet addr | grep -v -e 'inet 127\.' | head -1 | sed -e 's,^.*inet \([^/]\+\)/.*$,\1,g'
```
or, grab the IP of the interface of the default route:
```sh
defaultdev=$(ip route|grep ^default|tr ' ' '\n'|sed -e '/^$/d'|tail -1)
ip -o -f inet addr show $(defaultdev) | grep -v -e 'inet 127\.' | head -1 | sed -e 's,^.*inet \([^/]\+\)/.*$,\1,g'
```
still haven't finished a complete review of the PR, so maybe this is overkill if publicip only runs in-container